Services/Mail/classes/class.ilMailTemplateTableGUI.php

```diff
@@ -109,22 +115,24 @@ protected function fillRow($row)
 		}
 
 		$this->ctrl->setParameter($this->getParentObject(), 'tpl_id', $row['tpl_id']);
-		$actions = new ilAdvancedSelectionListGUI();
-		$actions->setListTitle($this->lng->txt('actions'));
-		$actions->setId('act_' . $row['tpl_id']);
-		if(count($this->contexts))
-		{
+		if($this->dic->rbac()->system()->checkAccess('write', $_GET['ref_id'])) {
```

Comment:
We should try to avoid accessing `$_GET` and use the HttpRequest from `$DIC` instead. I also suggest an (int) cast here.